### ISSUE TITLE
Template grouping

### ISSUE THREAD

**Issue by dg0yt** [OpenOrienteering/mapper#545]
(Split from #493)

**mlerjen reported on Sourceforge [[tickets:#493](https://sourceforge.net/p/oorienteering/tickets/493)]:**

Nice to have grouped templates, especially if working with tiled templates. Visibility and transparency could be modified more efficient as now.

![oo mapper notes_31](https://cloud.githubusercontent.com/assets/13567791/9567963/a69adf6c-4f3c-11e5-896e-ce6cfe12d76d.jpg)

**Comment by dominikzalewski** [comment]
I have given a lot of thought into it recently. Apart from that I'm also fighting with a problem that if you have too many too big templates on your tablet (while doing field work), the application just crashes. I was wondering if you could also try to optimize how templates are loaded. Namely, that if the template is hidden (or initially hidden) it is never loaded into memory. They would be loaded into memory only after I set them to visible.

If that was the case, I could easily figure out on tablet how many templates loaded is simply too much for a device to handle. Currently what I need to do is to delete almost all templates on the desktop from the map, load onto device and check if it works. If it is, add another template on the desktop, transfer to tablet, check. And so on. It's really painful and time consuming.

If it's too late to add on to this feature, let me know and I'll create a separate issue.

**Comment by dg0yt** [comment]
@dominikzalewski: Your comment is unrelated to template grouping. Open a separate issue.

**Comment by dominikzalewski** [comment]
Separate issue opened: https://github.com/OpenOrienteering/mapper/issues/598

**Comment by dominikzalewski** [comment]
Yes, I think that I would also benefit from grouping. Let me describe why. I map in Poland. Here we buy LIDAR data from a governmental institution. One 'rectangle' of LIDAR (around 1.25km^2 in rural areas) has a name that looks like this (an example):

N-33-142-B-b-1-3-2

And usually, to draw a map I have from a few to 10 or 20 of them. I generate various graphic files from LIDAR using lastools. So from one lidar rectangle I have for instance:
- hillshade of N-33-142-B-b-1-3-2
- ortophotoimage of N-33-142-B-b-1-3-2
- slopes of N-33-142-B-b-1-3-2
- infrared image of N-33-142-B-b-1-3-2
- iso lines of N-33-142-B-b-1-3-2

I'm putting them in different folders, but all the files have names N-33-142-B-b-1-3-2.png. But when I load them into OO Mapper, they all display just the name in the template dialog, not the full path (full path is only displayed when I hover over the name). So it's quite challenging to manage this number of templates. I tried to to merge all the images of one type (say hillshade) from various LIDAR rectangles into one using GDAL, but then I bumped on the problem that they wouldn't load on Android, because they are simply too big.

So my current workaround is to add a prefix or a suffix to the filename. As a result my template window looks more or less like this:

![screen shot 2016-02-26 at 10 13 08](https://cloud.githubusercontent.com/assets/13519572/13348063/9b9fafc6-dc71-11e5-9259-495f06c58893.png)

I think I would really like to have a way of grouping them.

**Comment by dominikzalewski** [comment]
> Nice to have the possibility to link single templates as groups to a switch button on the main screen. By clicking that button these linked templates can be switched on and off without proceeding through the template dialog. Useful when working with two templates simultaneously

Was thinking about exactly the same improvement the other day when doing fieldwork. Going through template dialog requires you to click three or more times. And there's so much space in the upper middle part of Android window to have rectangles named: T1, T2, T3, T4 (T - for template group) that you can just use as a pushbutton - once click.

But I am also aware that there's another issue in the log treating about rearrangement of Android menus, so a person picking this issue should also refer to the other one.
